### DISCUSSION TITLE
BootKeyboard on Nano 33 IoT

### DISCUSSION THREAD

**Discussion by dans1984** [NicoHood/HID#299] [Q&A]
Hi,

I think i'm needing to to use a Single Report Keyboard on my Nano 33 IoT. I need the device to report a HID device, rather than a Composite Device with a HID sub class.

I've tried to upload the boot keyboard sketch but get the following:
`libraries/HID-Project/HID-Project.a(BootKeyboard.cpp.o): In function `BootKeyboard_::getInterface(unsigned char*)':
/Users/user/Documents/Arduino/libraries/HID-Project/src/SingleReport/BootKeyboard.cpp:88: undefined reference to `USB_SendControl(unsigned char, void const*, int)'
libraries/HID-Project/HID-Project.a(BootKeyboard.cpp.o): In function `BootKeyboard_::send()':
/Users/user/Documents/Arduino/libraries/HID-Project/src/SingleReport/BootKeyboard.cpp:199: undefined reference to `USB_Send(unsigned char, void const*, int)'
libraries/HID-Project/HID-Project.a(BootKeyboard.cpp.o): In function `BootKeyboard_::getDescriptor(arduino::USBSetup&)':
/Users/user/Documents/Arduino/libraries/HID-Project/src/SingleReport/BootKeyboard.cpp:104: undefined reference to `USB_SendControl(unsigned char, void const*, int)'
libraries/HID-Project/HID-Project.a(BootKeyboard.cpp.o): In function `BootKeyboard_::setup(arduino::USBSetup&)':
/Users/user/Documents/Arduino/libraries/HID-Project/src/SingleReport/BootKeyboard.cpp:159: undefined reference to `USB_RecvControl(void*, int)'
/Users/user/Documents/Arduino/libraries/HID-Project/src/SingleReport/BootKeyboard.cpp:180: undefined reference to `USB_RecvControl(void*, int)'`

Is this particular function not supported on these boards? 

Many Thanks

**Comment by NicoHood** [comment]
The issue is tracked here:
https://github.com/NicoHood/HID/issues/257

As stated in my last comment, someone has to open an issue at the mentioned project. If you want support, open an issue there. Or even implement it, test it and open a PR. It should not be so hard, as it can be copy pasted from the other projects I guess.

**Reply by dans1984** [reply]
Its ok, I patched the code for this and some Arduino code and now it works fine.

**Reply by NicoHood** [reply]
Can you maybe share the code, that you have patched? It would help a lot of other people as well.
